serenata_toolbox/chamber_of_deputies/reimbursements.py

```diff
@@ -1,133 +1,39 @@
-        """
-        output = grouped[old].agg(func)
-        output = output.rename(index=new, inplace=True)
-        return output.reset_index()
-
-    def group(self, receipts):
-        log.info('Dropping rows without document_value or reimbursement_number…')
-        subset = ('document_value', 'reimbursement_number')
-        receipts = receipts.dropna(subset=subset)
-
-        groupby_keys = ('year', 'applicant_id', 'document_id')
-        receipts = receipts.dropna(subset=subset + groupby_keys)
-
-        receipts = receipts[receipts['document_value'] != 0]
-        receipts = receipts[receipts['reimbursement_number'] != '0']
-        receipts = receipts[receipts['year'] != 0]
-        receipts = receipts[receipts['applicant_id'] != '0']
-        receipts = receipts[receipts['document_id'] != '0']
-
-        log.info('Grouping dataset by applicant_id, document_id and year…')
-        grouped = receipts.groupby(groupby_keys)
-
-        log.info('Gathering all reimbursement numbers together…')
-        numbers = self.aggregate(
-            grouped,
-            'reimbursement_number',
-            'reimbursement_numbers',
-            lambda x: ', '.join(set(x))
-        )
-
-        log.info('Summing all net values together…')
-        net_total = self.aggregate(
-            grouped,
-            'net_value',
-            'total_net_value',
-            np.sum
-        )
-
-        log.info('Summing all reimbursement values together…')
-        total = self.aggregate(
-            grouped,
-            'reimbursement_value',
-            'reimbursement_value_total',
-            np.sum
-        )
-
-        log.info('Generating the new dataset…')
-        final = pd.merge(
-            pd.merge(pd.merge(total, net_total, on=groupby_keys), numbers, on=groupby_keys),
-            receipts,
-            on=groupby_keys
-        )
-        final = final.drop_duplicates(subset=groupby_keys)
-        final.rename(columns={'net_value': 'net_values',
-                              'reimbursement_value': 'reimbursement_values'},
-                     inplace=True)
-        final = final.drop('reimbursement_number', 1)
-        return final
-
-    @staticmethod
-    def unique_str(strings):
-        return ', '.join(set(strings))
-
-    def write_reimbursement_file(self, receipts):
-        log.info('Casting changes to a new DataFrame…')
-        df = pd.DataFrame(data=receipts)
-
-        log.info('Writing it to file…')
-        filepath = os.path.join(self.path, self.FILE_BASE_NAME)
-        df.to_csv(filepath, **self.CSV_PARAMS)
-
-        log.info('Done.')
+    """
+    Get an updated version of the reimbursements dataset for a given year.
+    """
+
+    def __init__(self, year, path=None):
+        self.year = year
+        self.path = path or gettempdir()
+
+    def __call__(self):
+        self.fetch()
+        self.clean()
+        file_path = os.path.join(
+            self.path, 'reimbursements-{}.csv'.format(self.year))
```

Comment:
As we're using Python 3.6 we could make it a bit more readable using the new f-strings:

```python
file_path = os.path.join(self.path, f'reimbursements-{self.year}.csv')
```